src/Tribe/Cache/Transient_Cache.php

```diff
@@ -25,13 +25,16 @@ public function reset_all( $post_id = 0 ) {
 		}
 
 		$hash = md5( serialize( $post_types ) );
-		foreach ( $this->keys as $key ) {
+		$is_using_object_cache = wp_using_ext_object_cache();
+
+		array_walk( $this->keys, function( $key ) use ( $hash, $is_using_object_cache ) {
 			delete_transient( __CLASS__ . $key );
 			delete_transient( __CLASS__ . $key . $hash );
-		}
-
-		// Flush WordPress cache for cache compatibility
-		wp_cache_flush();
+			if ( $is_using_object_cache ) {
```

Comment:
@codingmusician this is not flushing the cache unless you have a caching plugin, I'm wondering if we even need to make this check? As long as we're only deleting keys we have created ourselves like you're doing below, I don't see the need for this distinction.

Reply:
I was basing this solution on what was [mentioned here](https://github.com/the-events-calendar/event-tickets/pull/1673#issuecomment-634336682). Basically, `wp_cache_flush()` was [added](https://github.com/the-events-calendar/event-tickets/commit/ec3ca2e799dd7c9ca39f99355ca456fb38fa6d86) to ensure compatibility with object caching plugins.

Reply:
I supposed it would be safe just to go ahead and `wp_cache_delete()` all the keys, though. Is that what you're recommending @moraleida?

Reply:
yes @codingmusician. calling `wp_cache_delete()` on our keys will not cause issues if caching drop-ins are being used and it prevents weird issues with in-memory cache. We had just cast a net too wide with the earlier fix.